### PR TITLE
Bug fix/agency restart enhancements

### PR DESCRIPTION
**Pull request by kvahed**
- Removed unused Agent::active(...)
- Inception restart tolerates peer with missing active list
- UUIDs on disk outside database comparable with dbservers and coordinators

### CODE REVIEW

File: arangod/Agency/AgencyFeature.cpp [hunk 1]
```diff
@@ -128,6 +128,11 @@ void AgencyFeature::collectOptions(std::shared_ptr<ProgramOptions> options) {
                            "maximum size of appendEntries document (# log entries)",
                            new UInt64Parameter(&_maxAppendSize));
```
Comment (kvahed) [inline arangod/Agency/AgencyFeature.cpp]: Test

Comment (neunhoef) [inline arangod/Agency/AgencyFeature.cpp]: Noch einer.

Comment (neunhoef) [inline arangod/Agency/AgencyFeature.cpp]: Just a comment

File: arangod/Agency/Agent.cpp [hunk 2]
```diff
@@ -295,12 +295,16 @@ priv_rpc_ret_t Agent::recvAppendEntriesRPC(
   term_t term, std::string const& leaderId, index_t prevIndex, term_t prevTerm,
   index_t leaderCommitIndex, query_t const& queries) {
 
+  term_t t(this->term());
```
[the remark this answers is not in the document]
Comment (neunhoef) [reply arangod/Agency/Agent.cpp]: Nonsense.

File: arangod/Agency/Inception.cpp [hunk 3]
```diff
@@ -331,7 +332,7 @@ bool Inception::restartingActiveAgent() {
                 }
                 std::sort(myActVec.begin(),myActVec.end());
                 std::sort(theirActVec.begin(),theirActVec.end());
-                if (theirActVec != myActVec) {
+                if (theirActVec.size() > 1 && theirActVec != myActVec) {
                   if (!this->isStopping()) {
```
[the remark this answers is not in the document]
Comment (kvahed) [reply arangod/Agency/Inception.cpp]: fixed

File: arangod/Agency/Inception.cpp [hunk 4]
```diff
@@ -321,7 +322,7 @@ bool Inception::restartingActiveAgent() {
 
             if (i != active.end()) { // Member in my active list
               TRI_ASSERT(theirActive.isArray());
-              if (theirActive.length() == myActive.length()) {
+              if (theirActive.length() == 0 || theirActive.length() == myActive.length()) {
```
[the remark this answers is not in the document]
Comment (neunhoef) [reply arangod/Agency/Inception.cpp]: OK, we leave it for now.

**Pull request merged**